src/sim_engine/filesystem.rs

```diff
@@ -25,4 +25,11 @@ impl Filesystem for SimFilesystem {
         };
         Box::new(filesystem_copy)
     }
+    fn get_id(&self) -> String {
+        self.mount_point.clone()
+    }
+
+    fn eq(&self, other: &Filesystem) -> bool {
+        self.get_id() == other.get_id()
```

Comment:
This one works, because it is really equality. The fact that you can use ```==``` must be because String implements PartialEq. Is it possible to skip the cloning and do ```self.mountpoint == other.mountpoint```? I think it could work.